gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -141,7 +140,7 @@ else if (targetWidth <= font.data.spaceWidth) //
 					run.color.set(color);
 					run.x = x;
 					run.y = y;
-					fontData.getGlyphs(run, str, runStart, runEnd, colorRun);
+					fontData.getGlyphs(run, str, runStart, runEnd, false);
```

Comment:
`colorRun` was added in 8ed9d9d928cee9008fb055dc67a3588d0f15be8c. Can you take BitmapFontTest.java from that commit and see how it behaves with your changes versus without? Also see comment below about `xoffset` and BitmapFont `tightBounds`.

Reply:
1) With the original code
``fontData.getGlyphs(run, str, runStart, runEnd, colorRun);``
![glyphs2](https://user-images.githubusercontent.com/25801199/32607195-5c255f7c-c558-11e7-919d-3e1a1680d2b5.PNG)

2) With my code: always passing a false (dont using the tightBounds mode)
``fontData.getGlyphs(run, str, runStart, runEnd, false);``
![glyphs1](https://user-images.githubusercontent.com/25801199/32607194-5c0f02f4-c558-11e7-8401-a75da019eca8.PNG)

Maybe I'm missing something but as you commented on the issue #4192 
>...  Ideally colored and not colored text would have exactly the same spacing, but the complexity of breaking the text into runs makes it hard, especially when wrapping and truncation are considered.

Or there should be a new parameter on the getGlyphs function for this cases. I'm not sure.